proto/prysm/v2/slasher.proto

```diff
@@ -17,49 +30,104 @@ option php_namespace = "Ethereum\\Prysm\\v2";
 
 // Slasher service API
 //
-// Slasher service provides an interface for checking if attestations or blocks are slashable.
+// Slasher service provides an interface for checking if attestations or blocks
+// are slashable.
 service Slasher {
   // Returns any found attester slashings for an input indexed attestation.
-  rpc IsSlashableAttestation(ethereum.eth.v1alpha1.IndexedAttestation) returns (AttesterSlashingResponse) {
+  rpc IsSlashableAttestation(ethereum.eth.v1alpha1.IndexedAttestation)
+      returns (AttesterSlashingResponse) {
     option (google.api.http) = {
-      post: "/eth/v1alpha1/slasher/attestations/slashable",
-      body: "*"
+      post : "/eth/v1alpha1/slasher/attestations/slashable",
+      body : "*"
     };
   }
 
   // Returns any found proposer slashings for an input signed block header.
-  rpc IsSlashableBlock(ethereum.eth.v1alpha1.SignedBeaconBlockHeader) returns (ProposerSlashingResponse) {
+  rpc IsSlashableBlock(ethereum.eth.v1alpha1.SignedBeaconBlockHeader)
+      returns (ProposerSlashingResponse) {
     option (google.api.http) = {
-      get: "/eth/v1alpha1/slasher/blocks/slashable"
+      get : "/eth/v1alpha1/slasher/blocks/slashable"
     };
   }
 
-  // Returns the highest source and target attestation for validator indices that have been observed by slasher.
-  rpc HighestAttestations(HighestAttestationRequest) returns (HighestAttestationResponse) {
+  // Returns if a given indexed attestation could be slashable when compared to
+  // the slashers history for the attesters. This function is read-only, and
+  // does not need the indexed attestation to be signed.
+  rpc IsSlashableAttestationNoUpdate(ethereum.eth.v1alpha1.IndexedAttestation)
+      returns (Slashable) {
+    option deprecated = true;
+  };
+
+  // Returns if a given beacon block header could be slashable when compared to
+  // the slashers history for the proposer. This function is read-only, and does
+  // not need the beacon block header to be signed.
+  rpc IsSlashableBlockNoUpdate(ethereum.eth.v1alpha1.BeaconBlockHeader)
+      returns (Slashable) {
+    option deprecated = true;
+  };
+
+  // Returns the highest source and target attestation for validator indices
+  // that have been observed by slasher.
+  rpc HighestAttestations(HighestAttestationRequest)
+      returns (HighestAttestationResponse) {
     option (google.api.http) = {
-      get: "/eth/v1alpha1/slasher/attestations/highest"
+      get : "/eth/v1alpha1/slasher/attestations/highest"
     };
   }
 }
 
 message AttesterSlashingResponse {
-  ethereum.eth.v1alpha1.AttesterSlashing attester_slashing = 1;
+  repeated ethereum.eth.v1alpha1.AttesterSlashing attester_slashings = 1;
 }
 
 message ProposerSlashingResponse {
-  ethereum.eth.v1alpha1.ProposerSlashing proposer_slashing = 1;
+  repeated ethereum.eth.v1alpha1.ProposerSlashing proposer_slashings = 1;
 }
```

Comment:
Why changing these messages?

Reply:
Discussed offline, these changes bring over the current implementation message. The unreleased slasher will be amended to conform to these new messages.